hepcrawl/spiders/alpha_spider.py

```diff
@@ -0,0 +1,175 @@
+
+    def has_numbers(self, s):
+        """Detects if a string contains numbers"""
+        return any(char.isdigit() for char in s)
+
+    def parse_author_data(self, author_line):
+        """Parses the line where there are data about the author(s)"""
+        author_data = []
+        author_list = re.sub(r'[\n\t\xa0]', '', author_line).split(
+            ",")  # Author name might contain unwanted characters.
+
+        author = author_list[0]
+        surname, given_names = self.split_fullname(author)
+
+        for i in author_list:
+            if "thesis" in i.lower():
+                thesis_type = i.strip()
+            if "university" in i.lower():
+                affiliation = re.sub(r"[^A-Za-z\s]+", '', i).strip()
+            if self.has_numbers(i):
+                # Affiliation element might include the year
+                year = re.findall(r'\d+', i)[0].strip()
+
+        author_data.append({
+            'fullname': surname + ", " + given_names,
+            'surname': surname,
+            'given_names': given_names,
+            'thesis_type': thesis_type,
+            'affiliation': affiliation,
+            'year': year
+        })
+        return author_data
+
+    def get_authors(self, author_data):
+        """Gets the desired elements from author_data,
+        these will be put in the scrapy author item
+        """
+        authors = []
+        for author in author_data:
+            authors.append({
+                'surname': author['surname'],
+                'given_names': author['given_names'],
+                # 'full_name': author['fullname'],  # Not really necessary.
+                'affiliations': [{"value": author['affiliation']}]
+            })
+
+        return authors
+
+    def get_abstract(self, abs_pars):
+        """Returns a unified abstract, if divided to multiple paragraphs.
+        """
+        whole_abstract = " ".join(abs_pars)
+        return whole_abstract
+
+    def get_title(self, node):
+        title = node.xpath(
+            "./div[@class = 'node-headline clearfix']//a/text()").extract()
+        rel_url = node.xpath(
+            "./div[@class = 'node-headline clearfix']//a/@href").extract()
+        urls = [urljoin(self.domain, rel_url[0])]
+        return title, urls
+
+    def parse(self, response):
+        """Parse Alpha web page into a HEP record."""
+
+        # Random <br>'s will create problems
+        response = response.replace(body=response.body.replace('<br />', ''))
+        node = response.selector
+
+        for thesis in node.xpath(self.itertag):
+            record = HEPLoader(
+                item=HEPRecord(), selector=thesis, response=response)
+
+            # Author, affiliation, year:
+            author_line = thesis.xpath(
+                "./div[@class = 'content clearfix']//div[@class='field-item even']"
+                "/p[contains(text(),'Thesis')]/text()"
+            ).extract()
+            author_data = self.parse_author_data(author_line[0])
```

Comment:
Ah, I see now that the author_data contains more things. Hmm. Perhaps `parse_author_data` can return multiple arguments then. E.g. ```return authors, thesis_type, year```